pkg/assembler/backends/keyvalue/pkg.go

```diff
@@ -523,7 +525,173 @@ func hashVersionHelper(version string, subpath string, qualifiers map[string]str
 // Query Package
 
 func (c *demoClient) PackagesList(ctx context.Context, pkgSpec model.PkgSpec, after *string, first *int) (*model.PackageConnection, error) {
-	return nil, fmt.Errorf("not implemented: PackagesList")
+	c.m.RLock()
+	defer c.m.RUnlock()
+	if pkgSpec.ID != nil {
+		p, err := c.buildPackageResponse(ctx, *pkgSpec.ID, &pkgSpec)
+		if err != nil {
+			if errors.Is(err, errNotFound) {
+				// not found
+				return nil, nil
+			}
+			return nil, err
+		}
+
+		return &model.PackageConnection{
+			TotalCount: 1,
+			PageInfo: &model.PageInfo{
+				HasNextPage: false,
+				StartCursor: ptrfrom.String(p.ID),
+				EndCursor:   ptrfrom.String(p.ID),
+			},
+			Edges: []*model.PackageEdge{
+				{
+					Cursor: p.ID,
+					Node:   p,
+				},
+			},
+		}, nil
+	}
+
+	edges := make([]*model.PackageEdge, 0)
+	hasNextPage := false
+	numNodes := 0
+	totalCount := 0
+
+	if pkgSpec.Type != nil {
+		inType := &pkgType{
+			Type: *pkgSpec.Type,
+		}
+		pkgTypeNode, err := byKeykv[*pkgType](ctx, pkgTypeCol, inType.Key(), c)
+		if err == nil {
+			pNamespaces := c.buildPkgNamespace(ctx, pkgTypeNode, &pkgSpec)
+			for _, namespace := range pNamespaces {
+				for _, name := range namespace.Names {
+					for _, version := range name.Versions {
+						p := &model.Package{
+							ID:   pkgTypeNode.ThisID,
+							Type: pkgTypeNode.Type,
+							Namespaces: []*model.PackageNamespace{
+								{
+									ID:        namespace.ID,
+									Namespace: namespace.Namespace,
+									Names: []*model.PackageName{
+										{
+											ID:   name.ID,
+											Name: name.Name,
+											Versions: []*model.PackageVersion{
+												version,
+											},
+										},
+									},
+								},
+							},
+						}
+						edges = append(edges, &model.PackageEdge{
```

Comment:
same here, should check for `after` and `currentpage` like below

Reply:
I have added the check to all queries.